server/client.go

```diff
@@ -3118,7 +3226,10 @@ func (c *client) processSubsOnConfigReload(awcsti map[string]struct{}) {
 	for _, sub := range c.subs {
 		// Just checking to rebuild mperms under the lock, will collect removed though here.
 		// Only collect under subs array of canSubscribe and checkAcc true.
-		if !c.canSubscribe(string(sub.subject)) {
```

Comment:
I wonder if we should not have a `c.canSubscribe(sub)` instead that will do the check for the subject and internally check for queue perms? But maybe not.. route.go is calling canSubscribe() (as an alias to canExport()).

Reply:
Yes we were originally expanding `canSubscribe` but ended up separating it into `canQueueSubscribe` since being used for import/export permissions and that way `canSubscribe` can also be kept as a smaller function instead

Reply:
I agree with that, won't filter over routes w/ queues IMO.